tasks/main.yml

```diff
@@ -7,32 +7,15 @@
   user:
     name: "{{ newrelic_java_agent_user }}"
 
-- name: Create tmp dir for unarchiving
-  file:
-    path: "/tmp/newrelic-agent-install"
-    state: directory
-
-- name: Create the NR Java Agent directory
-  file:
-    path: "{{ newrelic_java_agent_path }}"
-    state: directory
-
 - name: Download the latest java agent .zip and unarchive it into NR Agent dir
   unarchive:
     remote_src: yes
-    dest: "/tmp/newrelic-agent-install"
+    dest: "{{ newrelic_java_agent_parent_path }}"
```

Comment:
is _newrelic_java_agent_parent_path_ declared elsewhere? (in an app-specific / deploy playbook or anything?) don't see it in the role defaults

i also think you may need a 'file' play prev. to this to create the dir before you can unarchive into it (unarchive req's it to exist first ... maybe? could be wrong)

Reply:
Yeah you're right 👍 Thanks for checking.  I had it set in the parent module. I've added the variables to the default vars and the file check.